org-code-javabuilder/theater/src/main/java/org/code/theater/Scene.java

```diff
@@ -0,0 +1,436 @@
+   * @param height the height of the ellipse
+   */
+  public final void drawEllipse(int x, int y, int width, int height) {
+    this.actions.add(
+        new DrawEllipseAction(
+            x, y, width, height, this.strokeColor, this.fillColor, this.strokeWidth));
+  }
+
+  /**
+   * Draws a rectangle on the canvas.
+   *
+   * @param x the left side of the rectangle
+   * @param y the top of the rectangle
+   * @param width the width of the rectangle
+   * @param height the height of the rectangle
+   */
+  public final void drawRectangle(int x, int y, int width, int height) {
+    this.actions.add(
+        new DrawRectangleAction(
+            x, y, width, height, this.strokeColor, this.fillColor, this.strokeWidth));
+  }
+
+  /**
+   * Sets the thickness of lines drawn. Note that this only applies to drawing operations performed
+   * by this scene.
+   *
+   * @param width width in pixels of the line to draw. Zero means no line.
+   */
+  public final void setStrokeWidth(double width) {
+    this.strokeWidth = width;
+  }
+
+  /**
+   * Sets the fill color for all shapes drawn. Note that this only applies to drawing operations
+   * performed by this scene.
+   *
+   * @param color the color value to fill any shape.
+   */
+  public final void setFillColor(Color color) {
+    this.fillColor = color;
+  }
+
+  /**
+   * Sets the color of lines drawn. Note that this only applies to drawing operations performed by
+   * this actor.
+   *
+   * @param color the color to draw lines with.
+   */
+  public final void setStrokeColor(Color color) {
+    this.strokeColor = color;
+  }
+
+  /**
+   * Sets the fill color for all shapes drawn. Note that this only applies to drawing operations
+   * performed by this actor.
+   *
+   * @param color the color name to fill any shape. If the name doesn't match a known color or hex
+   *     value, this call will set the fill color to nothing (e.g. transparent fill).
+   */
+  public final void setFillColor(String color) {
+    this.setFillColor(new Color(color));
+  }
+
+  /**
+   * Sets the color of lines drawn.
+   *
+   * @param color the color name to draw lines with. If the name doesn't match a known color or hex
+   *     value, this call will set the stroke color to nothing (e.g. transparent stroke). Note that
+   *     this only applies to drawing operations performed by this actor.
+   */
+  public final void setStrokeColor(String color) {
+    this.setStrokeColor(new Color(color));
+  }
+
+  /**
+   * Removes the stroke color so any shapes have no stroke. Note that this only applies to drawing
+   * operations performed by this actor.
+   */
+  public final void removeStrokeColor() {
+    this.strokeColor = null;
+  }
+
+  /**
+   * Removes the fill color so any shapes have no fill. Note that this only applies to drawing
+   * operations performed by this actor.
+   */
+  public final void removeFillColor() {
+    this.fillColor = null;
+  }
+
+  final List<SceneAction> getActions() {
```

Comment:
TIL: the default access modifier acts differently from public, private, and protected: https://www.baeldung.com/java-access-modifiers#:~:text=2.-,Default,package%20com.

Reply:
Mind blown. I was definitely about to ask for an explicit keyword here, but this is, in fact, the exact modifier needed. :)

Reply:
Oh interesting! I actually didn't know this either and just thought it defaulted to `protected`. Good to know!